### PR TITLE
[#236] Issues with deployment in failed state

### PR DESCRIPTION
**Pull request by JessikaCastellano**
**Description** 

Resolves: [#236](https://github.com/UffizziCloud/uffizzi_app/issues/236)

Deployment in `status=failed`

Message: **Deployment failed**

### CODE REVIEW

File: core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb [hunk 1]
```diff
@@ -97,7 +97,9 @@ def update
   # @response [object<errors: object<title: string>>] 404 Not found
   # @response 401 Not authorized
   def deploy_containers
-    deployment = resource_project.deployments.active.find(params[:id])
+    deployment = resource_project.deployments.find(params[:id])
```
Comment (zipofar) [inline core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb:100]: In `UffizziCore::DeploymentRepo` there is scope `existed`. I think we can use it here.

File: core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb [hunk 2]
```diff
@@ -174,4 +176,8 @@ def dependencies_params
   def render_invalid_file
     render json: { errors: { state: ['Invalid compose file'] } }, status: :unprocessable_entity
   end
+
+  def render_invalid_deployment_status(state)
+    render json: { errors: { title: ["Deployment #{state}"] } }, status: :not_found
```
Comment (zipofar) [inline core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb:181]: Let's use `unprocessable_entity`

File: core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb [hunk 3]
```diff
@@ -97,7 +97,11 @@ def update
   # @response [object<errors: object<title: string>>] 404 Not found
   # @response 401 Not authorized
   def deploy_containers
-    deployment = resource_project.deployments.active.find(params[:id])
+    deployment = resource_project.deployments.existed.find(params[:id])
+
+    unless deployment.active?
+      render json: { errors: { title: ["Deployment #{deployment.state}"] } }, status: :unprocessable_entity
+    end
```
Comment (zipofar) [inline core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments_controller.rb:104]: I think we didn't describe the bug clearly enough.

In this version code when we start update the previous failed preview we immediately see the error `Deployment failed`. It's not good from a user's point of view
Therefore we should not render this error here. We should just start update deployment.

There is one place where raise `Not Found` error
`core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/activity_items_controller.rb:18`

This action called by our CLI in cycle for check statuses. When our deployment gets status 'failed` this action raise `Not Found Deployment` error

File: core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/activity_items_controller.rb [hunk 4]
```diff
@@ -4,6 +4,8 @@ class UffizziCore::Api::Cli::V1::Projects::Deployments::ActivityItemsController
   UffizziCore::Api::Cli::V1::Projects::Deployments::ApplicationController
   before_action :authorize_uffizzi_core_api_cli_v1_projects_deployments_activity_items
 
+  rescue_from UffizziCore::DeploymentStateError, with: :render_deployment_state_exception
```
Comment (zipofar) [inline core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/activity_items_controller.rb:7]: I think it's over complicated. Let's do simple as possible.
In that case we can just write something like this:
```
  deployment = resource_deployment.existed
  if deployment.failed?
    return render json: { errors: { title: [I18n.t '...'] } }, status: :unprocessable_entity
  end

    activity_items = deployment
      .activity_items
      .page(page)
      .per(per_page)
      .order(updated_at: :desc)
      .ransack(q_param)
      .result
```

And let's use I18 everywhere possible

File: core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/events_controller.rb [hunk 5]
```diff
@@ -20,6 +20,11 @@ class UffizziCore::Api::Cli::V1::Projects::Deployments::EventsController <
   # @response 401 Not authorized
   # @response 404 Not found
   def index
+    unless resource_deployment.active?
```
Comment (zipofar) [inline core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/events_controller.rb:23]: Please add comment to issue with subscribe how QA can reproduce this case and get this error.

Comment (JessikaCastellano) [reply core/app/controllers/uffizzi_core/api/cli/v1/projects/deployments/events_controller.rb:23]: Added https://github.com/UffizziCloud/uffizzi_app/issues/236#issuecomment-1231788886

**Review by zipofar — APPROVED** [review]

**Pull request merged**